### PR TITLE
[school-test] Support for multiple runs per test instance

### PR DESCRIPTION
**Pull request by falkecarlsen**
Since each generation is seeded, it may render different results. To accurately test schools, multiple runs should be considered. This is enabled by this PR. Note that runtime is quite long, ~60 minutes for the shown example at 10 runs per test instance.

One interesting effect of upping the number of runs is that the mean divergence is reduced significantly. Now all cities results in a mean divergence below 1000m. This feels like a good result.

```
Executing school placement one-sided t-test on 10 runs of Esbjerg
	All schools placed closer than bound: False
	Mean divergence: 966.26 meters
	T-test with bound 1150 meters. T-stat: -4.30251, p-value: 0.00001
	Since t-stat is zero or negative, and p-value is of statistical significance (p <= 0.05), the null-hypothesis can be rejected.
Executing school placement one-sided t-test on 10 runs of Slagelse
	All schools placed closer than bound: True
	Mean divergence: 609.19 meters
	T-test with bound 1150 meters. T-stat: -16.73147, p-value: 0.00000
	Since t-stat is zero or negative, and p-value is of statistical significance (p <= 0.05), the null-hypothesis can be rejected.
Executing school placement one-sided t-test on 10 runs of Randers
	All schools placed closer than bound: False
	Mean divergence: 683.40 meters
	T-test with bound 1150 meters. T-stat: -14.53810, p-value: 0.00000
	Since t-stat is zero or negative, and p-value is of statistical significance (p <= 0.05), the null-hypothesis can be rejected.
Executing school placement one-sided t-test on 10 runs of Vejen
	All schools placed closer than bound: True
	Mean divergence: 522.59 meters
	T-test with bound 1150 meters. T-stat: -inf, p-value: 0.00000
	Since t-stat is zero or negative, and p-value is of statistical significance (p <= 0.05), the null-hypothesis can be rejected.
Executing school placement one-sided t-test on 10 runs of Aalborg
	All schools placed closer than bound: False
	Mean divergence: 797.18 meters
	T-test with bound 1150 meters. T-stat: -10.50443, p-value: 0.00000
	Since t-stat is zero or negative, and p-value is of statistical significance (p <= 0.05), the null-hypothesis can be rejected.
```

### CODE REVIEW

**Comment by falkecarlsen** [comment]
Note that the p-values are extremely low. I've considered if averaging p-values for a test on each run is better than one test for the entire list of divergences.

**Review by NicEastvillage — APPROVED** [review]

File: test_schools.py [hunk 1]
```diff
@@ -159,14 +170,29 @@ def run_test(test: TestInstance, bound: float, plot: bool):
     else:
         if p_val <= 0.05:
             print("\tThe t-stat is not zero or negative, and p-value is of statistical significance (p <= 0.05),"
-                  "the null-hypothesis cannot be rejected.")  # FIXME: does this mean that we can accept the null?
+                  "the null-hypothesis cannot be rejected.")
```
Comment (MathiasMehl) [inline test_schools.py]: When p <= 0.05, we can reject the null-hypothesis, not not reject

File: test_schools.py [hunk 2]
```diff
@@ -159,14 +170,29 @@ def run_test(test: TestInstance, bound: float, plot: bool):
...
         else:
             print("\tSince p-value is not of statistical significance, the null-hypothesis cannot be rejected.")
```
Comment (MathiasMehl) [inline test_schools.py:175]: In this case it is of statistical significance, and can be rejected

**Review by MathiasMehl — APPROVED** [review]

File: test_schools.py (hunk 1 above)
Comment (falkecarlsen) [reply test_schools.py]: Fixed in 76f779f

**Pull request merged**